### DISCUSSION TITLE
github-readme-stats grade

### DISCUSSION THREAD

**Discussion by DanX069** [anuraghazra/github-readme-stats#2752] [Q&A] [answered]
My github-readme-stats grade changed from A to B today, why? Please explain. I saw the calculation, there are only B+,  A,  A+, S, S+ grades then how B? And why did it reduce from A to B? You can check my profile and see all the stats.

**Comment by qwerty541** [comment, marked as answer]
Today ranking system was updated by merging #1186 into master. You can read about it in more detail in pull request description and comments.

**Reply by DanX069** [reply]
Ah, gotcha, thanks for letting me know, I realize I was just reading the formula part of calculateRank.js and did not know about the weight part.

**Reply by rickstaa** [reply]
@qwerty541 is right. I decided to merge #1186 since an improvement has been requested by many people. As explained in that issue, the old ranking function almost everybody A+ and made it nearly impossible to get a higher or lower ranking sweat_smile. As a result, @francois-rozet proposed a new ranking version that is more well-balanced and motivates people to contribute to OS project and follow good code practices. I know that it will not make everybody happy, but in my opinion, it results in a better ranking scheme.

Having that said, creating a good ranking scheme is hard when we don't have the ground truth mean of all developers on GitHub. As a result, the ranking mechanism is still open to change and therefore, feel free to suggest any improvements to the weights or calculation function in https://github.com/anuraghazra/github-readme-stats/issues/455 🙏🏻.
